lib/blocs/calendar_cubit.dart

```diff
@@ -122,88 +132,153 @@ class CalendarCubit extends Cubit<CalendarState> {
 
   /// The offset to be used for the next paginated request.
   int _nextOffset = 0;
+  int _nextPastOffset = 0;
+
+  /// The time rounded down to the month used to split the events in "past" and
+  /// "future". Also used in the query. This is a final to avoid race conditions
+  final DateTime _splitTime =
+      DateTime(DateTime.now().year, DateTime.now().month);
 
-  /// The time used as filter, stored so that later
-  /// paginated requests have the correct offset.
-  DateTime? _lastLoadTime;
+  /// A list of events that have been removed from the previous results
+  /// in order to prevent them filling up the calendar before today.
+  /// These should be added in later calls to [moreUp()].
+  final List<CalendarEvent> _remainingPastEvents = [];
 
   /// A list of events that have been removed from the previous results
   /// in order to prevent them filling up the calendar further then where
   /// the first not-loaded event will go later. These should be added in
   /// later calls to [more()].
-  final List<CalendarEvent> _remainingEvents = [];
+  final List<CalendarEvent> _remainingFutureEvents = [];
 
-  CalendarCubit(this.api) : super(const CalendarState.loading(results: []));
+  /// Debouncetimer to fix things like load
+  Timer? _debounce;
+
+  CalendarCubit(this.api) : super(const CalendarState.loading());
+
+  Future<void> cachedLoad() async {
+    if (_debounce == null || !_debounce!.isActive) {
+      await load();
+    }
+  }
 
   Future<void> load() async {
-    emit(state.copyWith(isLoading: true));
+    emit(const DoubleListState.loading());
+
+    _debounce = Timer(const Duration(minutes: 10), () => {});
+
     try {
-      _lastLoadTime = DateTime.now();
       final query = _searchQuery;
-      final start = query == null ? _lastLoadTime : null;
 
       // Get first page of events.
-      final eventsResponse = await api.getEvents(
-        start: start,
+      final futureEventsResponseFuture = api.getEvents(
+        start: _splitTime,
         search: query,
         ordering: 'start',
         limit: firstPageSize,
         offset: 0,
       );
+      // get -1st page
+      final pastEventsResponseFuture = api.getEvents(
+        end: _splitTime,
+        search: query,
+        ordering: '-end',
+        limit: pageSize,
+        offset: _nextPastOffset,
+      );
+
+      // Get all partner events.
+      final futurePartnerEventsResponseFuture = api.getPartnerEvents(
+        start: _splitTime,
+        search: query,
+        ordering: 'start',
+      );
+      // Get all partner events.
+      final pastPartnerEventsResponseFuture = api.getPartnerEvents(
+        start: _splitTime,
+        search: query,
+        ordering: 'start',
```

Comment:
shouldn't this be on -end? or at least -something, I'm not sure how end/start matters